Model/Sync/CollectionsProducts/Services/CollectionsProductsService.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Yotpo\Core\Model\Sync\CollectionsProducts\Services;
+
+use Yotpo\Core\Model\AbstractJobs;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Store\Model\App\Emulation as AppEmulation;
+use Yotpo\Core\Model\Config as CoreConfig;
+use Yotpo\Core\Model\Sync\Catalog\YotpoResource;
+
+class CollectionsProductsService extends AbstractJobs
+{
+    const PRODUCT_SYNC_LIMIT_CONFIG_KEY = 'product_sync_limit';
+    const YOTPO_COLLECTIONS_PRODUCTS_SYNC_TABLE_NAME = 'yotpo_collections_products_sync';
+
+    /**
+     * @var CoreConfig
+     */
+    protected $coreConfig;
+
+    /**
+     * @var YotpoResource
+     */
+    protected $yotpoResource;
+
+    /**
+     * @var int
+     */
+    protected $collectionsProductsSyncBatchSize = null;
+
+    /**
+     * Processor constructor.
+     * @param AppEmulation $appEmulation
+     * @param ResourceConnection $resourceConnection
+     * @param CoreConfig $coreConfig
+     * @param YotpoResource $yotpoResource
+     **/
+    public function __construct(
+        AppEmulation $appEmulation,
+        ResourceConnection $resourceConnection,
+        CoreConfig $coreConfig,
+        YotpoResource $yotpoResource
+    ) {
+        $this->coreConfig = $coreConfig;
+        $this->yotpoResource = $yotpoResource;
+        $this->collectionsProductsSyncBatchSize = $this->coreConfig->getConfig($this::PRODUCT_SYNC_LIMIT_CONFIG_KEY);
+        parent::__construct($appEmulation, $resourceConnection);
+    }
+
+    /**
+     * @param integer $collectionsProductsSyncBatchSize
+     * @return array<string>
+     */
+    public function getCollectionsProductsToSync($collectionsProductsSyncBatchSize) {
+        $storeId = $this->coreConfig->getStoreId();
+        $connection = $this->resourceConnection->getConnection();
+        $categoryProductsQuery = $connection->select(
+        )->from(
+            ['collection_product_entity' => $this->resourceConnection->getTableName($this::YOTPO_COLLECTIONS_PRODUCTS_SYNC_TABLE_NAME)],
```

Comment:
Please don't this alias, it's the same naming as Magento-native tables and it's super confusing.
You don't have to use alias here at all BTW.

Reply:
agree, removed it.